metals/src/main/scala/scala/meta/metals/Semanticdbs.scala

```diff
@@ -109,7 +110,17 @@ object Semanticdbs extends LazyLogging {
       cwd: AbsolutePath
   ): semanticdb3.TextDocuments = {
     val bytes = Files.readAllBytes(semanticdbPath.toNIO)
-    val sdb = semanticdb3.TextDocuments.parseFrom(bytes)
+    val sdb =
+      try {
+        semanticdb3.TextDocuments.parseFrom(bytes)
+      } catch {
+        case _: InvalidProtocolBufferException =>
+          logger.error(
+            s"Have you upgraded to SemanticDB v3? Error parsing $semanticdbPath"
```

Comment:
Does the protobuf have a specific version field we can check?

Reply:
After SemanticDB v3 there is TextDocument.schema :) I tried to implement a smarter thing to detect old payloads but saw the logic would have to be in the critical path, so I opted to catch exceptions instead.